src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax;
+
+import core.basesyntax.io.read.ReportReader;
+import core.basesyntax.io.read.impl.ReportReaderImpl;
+import core.basesyntax.io.read.impl.csv.CsvReaderImpl;
+import core.basesyntax.io.write.ReportWriter;
+import core.basesyntax.io.write.impl.ReportWriterImpl;
+import core.basesyntax.io.write.impl.csv.CsvWriterImpl;
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.convert.ReportConverter;
+import core.basesyntax.utils.convert.ReportConverterImpl;
+import core.basesyntax.utils.convert.csv.CsvConverterImpl;
+import core.basesyntax.utils.generate.ReportGenerator;
+import core.basesyntax.utils.generate.impl.ReportGeneratorImpl;
+import core.basesyntax.utils.generate.impl.csv.CsvGeneratorImpl;
+import core.basesyntax.utils.process.impl.ShopServiceImpl;
+import core.basesyntax.utils.transaction.FruitTransaction;
+import core.basesyntax.utils.transaction.handlers.OperationHandlerFactory;
+import java.io.FileNotFoundException;
+import java.util.List;
+
+public class Main {
+    private static final String CSV_FILE_PATH = "/";
+
+    public static void main(String[] args) {
+        ReportReader csvReader = new CsvReaderImpl(CSV_FILE_PATH);
+        String transactionsReport = readData(csvReader);
+        ReportConverter csvConverter = new CsvConverterImpl(transactionsReport);
+        List<FruitTransaction> transactions = convertData(csvConverter);
+        OperationHandlerFactory operationHandlerFactory = new OperationHandlerFactory();
+        StorageImpl storage = new StorageImpl();
+        ShopServiceImpl shop = new ShopServiceImpl(operationHandlerFactory, storage);
+        shop.process(transactions);
+        ReportGenerator csvGenerator = new CsvGeneratorImpl();
+        ReportGenerator reportGenerator = new ReportGeneratorImpl(csvGenerator);
+        String report = reportGenerator.getReport(storage);
+        ReportWriter csvWriter = new CsvWriterImpl();
+        ReportWriter reportWriter = new ReportWriterImpl(csvWriter);
+        reportWriter.write(report, CSV_FILE_PATH);
```

Comment:
According to the checklist, all services should be independent and called in the main() method. The instantiation and usage of services in the main method should be simplified and not nested. For example, rather than creating a ReportReader and immediately calling a method on it, you should instantiate all necessary services first and then call their methods one by one.